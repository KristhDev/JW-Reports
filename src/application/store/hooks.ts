import { useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch = useDispatch.withTypes<AppDispatch>();
export const useAppSelector = useSelector.withTypes<RootState>();