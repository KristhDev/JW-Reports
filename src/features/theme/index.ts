export { default as themeReducer, setTheme } from './slice';