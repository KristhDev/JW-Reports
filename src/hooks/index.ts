export { default as useAuth } from './useAuth';
export { default as useTheme } from './useTheme';