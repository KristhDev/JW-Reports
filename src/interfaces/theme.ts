export type Theme = 'dark' | 'light' | 'default';

/* Defining the interface for the ThemeState. */
export interface ThemeState {
    colors: Colors;
    deviceTheme: Theme;
    isLoadedTheme: boolean;
    selectedTheme: Theme,
    theme: Theme,
}

/* Defining the interface for the Colors object. */
export interface Colors {
    background: string;
    bottom: string;
    button: string;
    buttonDark: string;
    card: string;
    contentHeader: string;
    focus: string;
    header: string;
    headerText: string;
    icon: string;
    inputText: string;
    linkText: string;
    modal: string;
    modalText: string;
    navbar: string;
    text: string;
    titleSecondary: string;
    titleText: string;
}

/**
 * SetThemePayload is an object with a property called theme that is of type Theme
 * @property {Theme} theme - The theme to set.
 */
export type SetThemePayload = {
    theme: Theme
}