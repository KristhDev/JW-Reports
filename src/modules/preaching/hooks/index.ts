export { default as usePreaching } from './usePreaching';