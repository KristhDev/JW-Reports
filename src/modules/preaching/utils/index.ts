export * from './constants';
export * from './report';