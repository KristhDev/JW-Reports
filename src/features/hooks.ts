import { TypedUseSelectorHook, useDispatch, useSelector } from 'react-redux';

/* Store */
import { AppDispatch, RootState } from './store';

/* Defined types hooks */
export const useAppDispatch: () => AppDispatch = useDispatch;
export const useAppSelector: TypedUseSelectorHook<RootState> = useSelector;