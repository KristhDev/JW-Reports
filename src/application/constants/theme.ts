import { ItemOption } from '../../infrasturcture/interfaces';

/**
 * The theme options to select
 */
export const THEME_OPTIONS: ItemOption[] = [
    { label: 'Modo predeterminado', value: 'default' },
    { label: 'Modo claro', value: 'light' },
    { label: 'Modo oscuro', value: 'dark' }
];