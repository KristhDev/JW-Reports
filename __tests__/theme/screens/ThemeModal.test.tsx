import React from 'react';
import { fireEvent, render, screen } from '@testing-library/react-native';
import hexToRgba from 'hex-to-rgba';

import { ThemeModal } from '../../../src/theme';

import { useTheme } from '../../../src/hooks';

import { THEME_OPTIONS } from '../../../src/utils';

import { darkColors } from '../../../src/theme';

const onCloseMock = jest.fn();
const setThemeMock = jest.fn();

jest.mock('../../../src/hooks/useTheme.ts')

describe('Test in <ThemeModal /> screen', () => {
    (useTheme as jest.Mock).mockReturnValue({
        state: { colors: darkColors, theme: 'dark' },
        BUTTON_TRANSLUCENT_COLOR: hexToRgba(darkColors.button, 0.50),
        setTheme: setThemeMock
    });

    beforeEach(() => {
        render(
            <ThemeModal
                isOpen
                onClose={ onCloseMock }
            />
        );

        jest.clearAllMocks();
    });

    it('should to match snapshot', () => {
        expect(screen.toJSON()).toMatchSnapshot();
    });

    it('should call setTheme with respective value', () => {
        const options = screen.getAllByTestId('radio-btn-text');
        fireEvent.press(options[2]);

        expect(setThemeMock).toHaveBeenCalledTimes(1);
        expect(setThemeMock).toHaveBeenCalledWith(THEME_OPTIONS[2].value);
    });
});