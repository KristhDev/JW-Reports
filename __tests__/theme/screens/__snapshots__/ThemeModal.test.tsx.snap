// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <ThemeModal /> screen should to match snapshot 1`] = `
<Modal
  animationType="fade"
  hardwareAccelerated={false}
  statusBarTranslucent={true}
  transparent={true}
  visible={true}
>
  <View
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(0, 0, 0, 0.5)",
        "flex": 1,
        "justifyContent": "center",
        "zIndex": 999,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#292929",
          "borderRadius": 10,
          "justifyContent": "space-between",
          "padding": 15,
          "width": "87%",
        }
      }
    >
      <View
        style={
          Array [
            Object {
              "padding": 10,
            },
            Object {
              "marginBottom": 10,
              "marginLeft": 5,
              "padding": 0,
            },
          ]
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "#BFBFBF",
                "fontSize": 17,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
                "fontSize": 20,
                "fontWeight": "bold",
                "textAlign": "left",
              },
            ]
          }
          testID="info-text-text"
        >
          Apariencia
        </Text>
      </View>
      <View
        accessibilityRole="radiogroup"
      >
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 5,
            }
          }
          testID="radio-btn-container"
        >
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              Object {
                "checked": false,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            color="#C0A7E1"
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                false,
                Object {
                  "borderRadius": 18,
                  "padding": 6,
                },
              ]
            }
            uncheckedColor="#BFBFBF"
          >
            <View
              style={
                Object {
                  "opacity": 0,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#C0A7E1",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "lineHeight": 24,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                󰄬
              </Text>
            </View>
          </View>
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "marginLeft": 15,
              }
            }
            testID="radio-btn-text"
          >
            Modo predeterminado
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 5,
            }
          }
          testID="radio-btn-container"
        >
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              Object {
                "checked": false,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            color="#C0A7E1"
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                false,
                Object {
                  "borderRadius": 18,
                  "padding": 6,
                },
              ]
            }
            uncheckedColor="#BFBFBF"
          >
            <View
              style={
                Object {
                  "opacity": 0,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#C0A7E1",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "lineHeight": 24,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                󰄬
              </Text>
            </View>
          </View>
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "marginLeft": 15,
              }
            }
            testID="radio-btn-text"
          >
            Modo claro
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "flexDirection": "row",
              "padding": 5,
            }
          }
          testID="radio-btn-container"
        >
          <View
            accessibilityLiveRegion="polite"
            accessibilityRole="radio"
            accessibilityState={
              Object {
                "checked": true,
                "disabled": false,
              }
            }
            accessible={true}
            collapsable={false}
            color="#C0A7E1"
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Array [
                Object {
                  "overflow": "hidden",
                },
                false,
                Object {
                  "borderRadius": 18,
                  "padding": 6,
                },
              ]
            }
            uncheckedColor="#BFBFBF"
          >
            <View
              style={
                Object {
                  "opacity": 1,
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "#C0A7E1",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "lineHeight": 24,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                󰄬
              </Text>
            </View>
          </View>
          <Text
            onPress={[Function]}
            style={
              Object {
                "color": "#FFFFFF",
                "fontSize": 18,
                "marginLeft": 15,
              }
            }
            testID="radio-btn-text"
          >
            Modo oscuro
          </Text>
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "justifyContent": "flex-end",
            "marginTop": 20,
          }
        }
      >
        <View
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessible={true}
          collapsable={false}
          focusable={true}
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Array [
                Object {
                  "backgroundColor": "#C0A7E1",
                  "borderRadius": 10,
                },
                Object {
                  "backgroundColor": "transparent",
                  "marginRight": 5,
                },
              ],
            ]
          }
          testID="button-touchable"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "flexDirection": "row",
                  "justifyContent": "center",
                  "paddingHorizontal": 20,
                  "paddingVertical": 8,
                },
                Object {
                  "paddingHorizontal": 12,
                },
              ]
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "textAlign": "left",
                  },
                  Object {
                    "color": "rgba(28, 27, 31, 1)",
                    "fontFamily": "System",
                    "fontWeight": "400",
                    "letterSpacing": 0,
                  },
                  Object {
                    "writingDirection": "ltr",
                  },
                  Array [
                    Object {
                      "color": "#121212",
                      "fontSize": 18,
                      "fontWeight": "bold",
                      "marginBottom": 2,
                      "textAlign": "center",
                    },
                    Object {
                      "color": "#C0A7E1",
                      "fontSize": 16,
                    },
                  ],
                ]
              }
              testID="button-text"
            >
              CANCELAR
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</Modal>
`;
